src/main/java/seedu/agendum/model/ModelManager.java

```diff
@@ -82,6 +82,7 @@ public ReadOnlyToDoList getToDoList() {
 
     /** Raises an event to indicate the model has changed */
     private void indicateToDoListChanged() {
+        toDoList.resetData(toDoList);
```

Comment:
I see, I think the right way to do this would be to use some sort of 'diffing' algorithm to compare what's the difference between the data before and after inserting an item and just perform the row insertion/deletion at the parts where the difference occurs. However, this would be too complicated for our purposes here.

If the performance hit of doing this is not too bad, let's just stay with this?